LibSerialize.lua

```
@@ -1315,25 +1466,52 @@ function LibSerialize:SerializeEx(opts, ...)
 
     ser:_WriteByte(SERIALIZATION_VERSION)
 
-    for i = 1, select("#", ...) do
-        local input = select(i, ...)
-        if not ser:_WriteObject(input) then
-            -- An unserializable object was passed as an argument.
-            -- Write nil into its slot so that we deserialize a
-            -- consistent number of objects from the resulting string.
-            ser:_WriteObject(nil)
+    local operation = function(...)
+        for i = 1, select("#", ...) do
+            local input = select(i, ...)
+            if not ser:_WriteObject(input) then
+                -- An unserializable object was passed as an argument.
+                -- Write nil into its slot so that we deserialize a
+                -- consistent number of objects from the resulting string.
+                ser:_WriteObject(nil)
+            end
+        end
+        return ser._flushWriter()
+    end
+
+    if ser._opts.asyncMode then
+        local thread = coroutine_create(operation)
+        local dots = {...}
+        -- return coroutine handler
+        return function()
+            local co_success, result = coroutine_resume(thread, unpack(dots))
+            if not co_success then
+                error('Coroutine failed')
```

You need to re-throw the actual error, otherwise it's not debuggable.